src/test/kotlin/blackjack/domain/SortedShuffleStrategy.kt

```diff
@@ -0,0 +1,10 @@
+package blackjack.domain
+
+import blackjack.domain.card.Card
+import blackjack.domain.card.ShuffleStrategy
+
+class SortedShuffleStrategy : ShuffleStrategy {
```

Comment:
랜덤이 아닌 구현체를 만들어 유연하게 테스트를 진행해주셨네요.

현재 비즈니스 로직 상 `ShuffleStrategy`의 구현체가 하나만 정의되어 있네요.

구조상 인터페이스 추출을 통한 추상화를 통해 이점이 있다고 판단해서 진행해주신거라면 괜찮을 것 같은데, 단지 테스트만을 위해 인터페이스를 추출하신거라면 조금은 고민이 필요할 것 같습니다.

추가로 코틀린에서 람다를 통해 인터페이스 구현하는 부분에 대해 문서 남겨두었으니 참고 부탁드립니다. (1.4 버전부터 사용이 가능합니다.)

```
Unlike Java, Kotlin will not allow a SAM conversion for every interface with one single abstract method. We believe that an intention to make an interface applicable for SAM conversion should be explicit. Thus, to define a SAM interface, you’ll need to mark an interface with the fun keyword to emphasize that it can be used as a functional interface:
```

[https://blog.jetbrains.com/kotlin/2019/12/what-to-expect-in-kotlin-1-4-and-beyond/#sam-conversions](https://blog.jetbrains.com/kotlin/2019/12/what-to-expect-in-kotlin-1-4-and-beyond/#sam-conversions)

Reply:
테스트만을 위한 인터페이스 추출이었고, 말씀해주신 적용해서 변경했습니다.
다만 이 책임을 어디에 줄지 고민이네요.
테스트와 확장성을 고려해서 셔플 로직을 끌어 올린 만큼, 객체 그래프의 상위 부분에서 관리하는 것이 적절하다고 생각이 드는데, 막상 함수가 담고 있는 부분은 Card에 의존하고 있어서 Util성으로 활용해야겠다고 생각했고, 클래스마저 필요 없다고 생각해서 상수로 관리하도록 했습니다.
어색한 부분이 있다면 지적 부탁드립니다.

Reply:
Kotlin의 경우 Functional Interface에 대해 람다로 심플하게 구현하는 부분이 잘 지원되지 않아서 해당 부분에 대해 코멘트를 남겨두었었습니다.

미션의 내용과 별개로 코멘트를 남겨드렸던 건데 조금 오해가 있었네요 ㅠㅠ High Order Function 형태로 변경을 의도한 내용은 아니었어요.

High Order Function으로 변경해주셨는데 해당 함수의 반환 타입이 Unit이 아니라 명확히 반환 타입이 정해진 High Order Function이라 크게 어색하거나 이상하게 느껴지진 않네요!